Structure_Engine/Query/Geometry.cs

```diff
@@ -72,10 +78,15 @@ public static IGeometry Geometry(this RigidLink link)
         {
             List<IGeometry> lines = new List<IGeometry>();
 
-            foreach (Node sn in link.SecondaryNodes)
+            if (link?.SecondaryNodes != null && link?.PrimaryNode?.Position != null)
```

Comment:
Possibly we could have an `else` statement here returning `null`, to make sure that we distinguish between ampty and corrupt geometry?

Reply:
Same as above. Given that we are returning a `CompositeGeometry`, its good to be consistent.